Src/CSharpier/SyntaxPrinter/SyntaxNodePrinters/ForEachStatement.cs

```diff
@@ -20,34 +20,36 @@ public static Doc Print(ForEachStatementSyntax node)
                 ExtraNewLines.Print(node),
                 leadingTrivia,
                 Doc.Group(
-                    Token.PrintWithoutLeadingTrivia(node.AwaitKeyword),
-                    node.AwaitKeyword.Kind() != SyntaxKind.None ? " " : Doc.Null,
-                    node.AwaitKeyword.Kind() == SyntaxKind.None
-                        ? Token.PrintWithoutLeadingTrivia(node.ForEachKeyword)
-                        : Token.Print(node.ForEachKeyword),
-                    " ",
-                    Token.Print(node.OpenParenToken),
-                    Doc.GroupWithId(
-                        groupId,
-                        Doc.Indent(
-                            Doc.SoftLine,
-                            Node.Print(node.Type),
-                            " ",
-                            Token.Print(node.Identifier),
-                            " ",
-                            Token.Print(node.InKeyword),
-                            " ",
-                            Node.Print(node.Expression)
+                    Doc.Group(
+                        Token.PrintWithoutLeadingTrivia(node.AwaitKeyword),
+                        node.AwaitKeyword.Kind() != SyntaxKind.None ? " " : Doc.Null,
```

Comment:
Have you considered if this whole thing might look better with pattern matching?

Reply:
Pattern matching is still newish to me so I wasn't even thinking about using it here. My thoughts on doing it here after I changed it.
The `is not` is more typing, but I don't mind the way it looks and it kinda reads better. Although changing it caused csharpier to break in an ugly way. I'm thinking csharpier should probably break it like so instead
```
node.AwaitKeyword.Kind() is not SyntaxKind.None 
    ? " " : Doc.Null, 
// or maybe
node.AwaitKeyword.Kind() is not SyntaxKind.None 
    ? " "
    : Doc.Null, 
```

I read a discussion the other day where someone pointed out that pattern matching is really nice for something like
```
if (!(someObject is SomeType))
// vs
if (someObject is not SomeType)
```
I did use pattern matching for gross condition in another issue just now, and I'm starting to drink the pattern matching kool-aid.

Reply:
When I said pattern matching, I meant the funky new match expression
```cs
  " ",
  node.AwaitKeyword.Kind() switch {                                                                                                                                                                                                            
      not SyntaxKind.None => Token.Print(node.ForEachKeyword),                                                                                                                                                                                 
      SyntaxKind.None => Token.PrintWithoutLeadingTrivia(node.ForEachKeyword)                                                                                                                                                                  
  },
  " ",
```

But at the time I thought the new instance of the ternary operator was a nested condition. Looking at the code in more detail I see that they are in fact separate arguments to the `Doc.Group()` call. Thus, I don't think there's a significantly nicer way of expressing that than your original code.